src/back/hlsl/writer.rs

```diff
@@ -1204,8 +1204,53 @@ impl<'a, W: fmt::Write> super::Writer<'a, W> {
                 self.temp_access_chain = chain;
                 self.named_expressions.insert(result, res_name);
             }
-            Statement::Switch { .. } => {
-                return Err(Error::Unimplemented(format!("write_stmt {:?}", stmt)))
+            Statement::Switch {
+                selector,
+                ref cases,
+                ref default,
+            } => {
+                // Start the switch
+                write!(self.out, "{}", INDENT.repeat(indent))?;
+                write!(self.out, "switch(")?;
+                self.write_expr(module, selector, func_ctx)?;
+                writeln!(self.out, ") {{")?;
+
+                // Write all cases
+                let indent_str_1 = INDENT.repeat(indent + 1);
+                let indent_str_2 = INDENT.repeat(indent + 2);
+
+                for case in cases {
+                    writeln!(self.out, "{}case {}: {{", &indent_str_1, case.value)?;
+
+                    for sta in case.body.iter() {
+                        self.write_stmt(module, sta, func_ctx, indent + 2)?;
+                    }
+
+                    if case.fall_through {
+                        writeln!(self.out, "{}/* fallthrough */", &indent_str_2)?;
+                        writeln!(self.out, "{}{{", &indent_str_2)?;
+                        writeln!(self.out, "{}break;", INDENT.repeat(indent + 3))?;
```

Comment:
I don't understand. This goes into `break` and the other branch goes into `break` too?

Reply:
Sorry, misunderstood `tint` logic in fallthrough case. We should generate it in new block.